src/region.c

```diff
@@ -337,6 +461,33 @@ gdip_getlowestrect (GpRectF *rects, int cnt, GpRectF* src, GpRectF* rslt)
 	return TRUE;
 }
 
+/* Finds a rect that has the lowest x and y after the src rect provided */
+static BOOL
+gdip_getlowestrect_sorted (GpRectF *rects, int cnt, int* index, GpRectF* src, GpRectF* rslt)
+{
+	GpRectF *current;
+	BOOL found = FALSE;
+
+	for (current = rects + *index; *index < cnt; (*index)++, current++) {
+		if (current->Width <= 0 || current->Height <= 0)
+			continue;
+
+		if (current->Y > src->Y ||
+			(current->Y == src->Y && current->X > src->X)) {
+			found = TRUE;
+			break;
+		}
```

Comment:
Won't this always be true?

Or, put another way, how are two rectangles with the same origin handled? Eg. merging {4, 4; 100x100} with {4, 4; 200x200}.

Reply:
Yes, assuming nothing sets a rectangle's location to (0, 0) (something that union used to do, which I've removed) and then sorts the array.

As long as there are only two rectangles with the same origin it is fine, and it shouldn't be possible to have more than two (would require a region made from specially-crafted region data). This'll get the first, and then the middle loop will get the second one. But still something worthy of thought.

Reply:
Right, so the same origin case is handled properly. Then it looks like this check should not be necessary anymore.

Reply:
You can actually get rid of `gdip_getlowestrect_sorted` altogether and replace it with simple loop like this: https://github.com/filipnavara/libgdiplus/commit/e3296d45ac07d6e4a1a68444430b07fb21da6ec0

Reply:
That looks excellent. How do we include that?